src/sctools/metrics/aggregator.py

```diff
@@ -455,8 +455,12 @@ def parse_extra_fields(self, tags: Sequence[str], record: pysam.AlignedSegment)
             self._quality_above_threshold(
                 30, self._quality_string_to_numeric(record.get_tag(consts.QUALITY_CELL_BARCODE_TAG_KEY))))
 
-        self.perfect_cell_barcodes += (
-            record.get_tag(consts.RAW_CELL_BARCODE_TAG_KEY) == record.get_tag(consts.CELL_BARCODE_TAG_KEY))
+        try:
```

Comment:
Minor suggestion: break up line 459 into two lines, only cover the `get_tag` call that should trigger the key error with the `try: except` to ensure we don't also catch all the reads in case we receive data whose tags don't match our other constant for some reason.